lib/harvester.js

```diff
@@ -197,6 +200,29 @@ Harvester.prototype._preprocessSchema = function (schema) {
   return schema;
 };
 
+Harvester.prototype._convertToLegacySchema = function(schema) {
+  var legacySchema = {};
+  var schemaMap = {
+    'string' : String,
+    'number' : Number,
+    'date' : Date,
+    'buffer' : Buffer,
+    'boolean' : Boolean,
+    'array' : Array
+  };
+
+  _.each(schema, function(item, key) {
+    if(item.isJoi) {
+      legacySchema[key] = schemaMap[item._type];
+    } else {
+      legacySchema[key] = item;
```

Comment:
I think what you mean here is taken care of by ```item.isJoi``` check, which the links parameters fails and gets copied over as is?